src/app/_components/BlockList/Sockets/useSubscribeBlocks.ts

```diff
@@ -35,6 +35,10 @@ export function useSubscribeBlocks(
       disconnect?.();
```

Comment:
call the unsubscribe here as well
 subscription.current.unsubscribe();
        subscription.current = undefined;